foundation/foundation-mda/src/test/java/com/boozallen/aiops/mda/metamodel/element/AbstractModelInstanceSteps.java

```diff
@@ -284,17 +284,28 @@ protected void createProject(String projectName, String moduleName, String url)
         writePom(model, projectDir);
     }
 
-    protected void createPipeline(String name, String typeName, String implName) throws IOException {
-        createPipeline(name, typeName, implName, p -> {});
+    protected void createAndSavePipeline(String name, String typeName, String implName) throws IOException {
+        createAndSavePipeline(name, typeName, implName, p -> {});
     }
 
-    protected void createPipeline(String name, String typeName, String implName, Consumer<PipelineElement> customization) throws IOException {
+    protected void createAndSavePipeline(String name, String typeName, String implName, Consumer<PipelineElement> customization) throws IOException {
         this.pipeline = TestMetamodelUtil.createPipelineWithType(name, BOOZ_ALLEN_PACKAGE, typeName, implName);
         customization.accept(pipeline);
         savePipelineToFile(pipeline);
         pipelines.put(name, pipeline);
     }
 
+    protected PipelineElement createPipeline(String name, String typeName, String implName) throws IOException {
+        this.pipeline = TestMetamodelUtil.createPipelineWithType(name, BOOZ_ALLEN_PACKAGE, typeName, implName);
+        pipelines.put(name, pipeline);
+        return pipeline;
+    }
+
+    protected void saveFileStores(List<FileStore> fileStores) throws IOException  {
+        this.pipeline.setFileStores(fileStores);
+        savePipelineToFile(pipeline);
+    }
```

Comment:
Refactored.

Reply:
Since `SparkApplicationStep extends AbstractModelInstanceSteps`, you should be able to access it with just `this.pipeline` within ` SparkApplicationStep`

Reply:
Turns out I had defined a conflicting pipeline element, removing it did the trick. Should be resolved.